subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -167,25 +149,31 @@ int aws_fota_parse_update_job_exec_state_rsp(char *update_rsp_document,
 	return ret;
 }
 
-int aws_fota_parse_notify_next_document(char *job_document,
-		u32_t payload_len, char *job_id_buf, char *hostname_buf,
-		char *file_path_buf)
+int aws_fota_parse_job_execution(char *job_document, u32_t payload_len,
+				 char *job_id_buf, char *hostname_buf,
+				 char *file_path_buf, int *version_number)
 {
 	struct notify_next_obj job;
 	struct job_document_obj *job_doc_obj;
 
-	int ret = json_obj_parse(job_document,
-				 payload_len,
+	int ret = json_obj_parse(job_document, payload_len,
 				 notify_next_obj_descr,
-				 ARRAY_SIZE(notify_next_obj_descr),
-				 &job);
-	job_doc_obj = &job.execution.job_document;
+				 ARRAY_SIZE(notify_next_obj_descr), &job);
+
+	if (!(ret & BIT(2))) {
```

Comment:
IMO 2 should have a define. Also, should there be a BUILD_ASSERT that 2 is always the right number?

Reply:
2 Can have a define but I have no idea what it should be called `EXECUTION_OBJECT_FIELD` ?

I don't know if we can create a build assert 

See:
https://docs.zephyrproject.org/latest/reference/misc/index.html?highlight=json#_CPPv414json_obj_parsePc6size_tPK14json_obj_descr6size_tPv

So it basically means that it successfully decoded something into the 3rd element in the desc_job_execution_obj structure. I can't really think of a way this can change other than AWS changing how they structure their data.